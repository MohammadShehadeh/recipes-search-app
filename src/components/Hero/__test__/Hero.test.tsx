import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Hero } from '..';

describe('Hero', () => {
  it('should render to match snapshot', () => {
    const tree = renderer.create(<Hero />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Hero correctly', () => {
    render(<Hero />);
    const heroElement = screen.getByText(
      /Discover Culinary Excellence 🍽️ Your Gateway to Limitless Recipes!/i
    );
    expect(heroElement).toBeInTheDocument();
  });
});
