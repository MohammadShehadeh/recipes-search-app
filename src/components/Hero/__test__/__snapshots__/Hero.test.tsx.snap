// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should render to match snapshot 1`] = `
Array [
  <section
    className="hero"
  >
    <div
      className="background"
      style={
        Object {
          "backgroundImage": "url('/food-hero.jpg')",
        }
      }
    >
      <div
        className="container"
      >
        <div
          className="wrapper main"
        >
          <span
            className="eyebrow contained"
          >
            <span
              className="label"
            >
              Healthy Lifestyle
            </span>
          </span>
          <h1
            className="headline"
          >
            Discover Culinary Excellence üçΩÔ∏è Your Gateway to Limitless Recipes!
          </h1>
        </div>
      </div>
    </div>
  </section>,
  <svg
    className="separator"
    preserveAspectRatio="none"
    viewBox="0 0 1440 74"
  >
    <path
      d="M456.464 0.0433865C277.158 -1.70575 0 50.0141 0 50.0141V74H1440V50.0141C1440 50.0141 1320.4 31.1925 1243.09 27.0276C1099.33 19.2816 1019.08 53.1981 875.138 50.0141C710.527 46.3727 621.108 1.64949 456.464 0.0433865Z"
    />
  </svg>,
]
`;
